test/sqllogictest/index_planning.slt

```diff
@@ -120,3 +140,84 @@ this
 -4.400
 still
 0.000
+
+statement ok
+drop index foo_idx2;
+
+#indexes on bar(a), foo(a), and foo(b) exist
+query T multiline
+explain plan for select foo.a, b, c, d, e from foo, bar where foo.a = bar.a and b = 'this'
+----
+Project {
+  outputs: [0 .. 2, 5, 6],
+  Join {
+    variables: [[(0, 0), (2, 0)], [(1, 0), (2, 1)]],
+    implementation: DifferentialLinear,
+    Get { materialize.public.bar (u3) },
+    ArrangeBy { keys: [[]], Constant [["this"]] },
+    ArrangeBy {
+      keys: [[#0, #1]],
+      Get { materialize.public.foo (u1) }
+    }
+  }
+}
```

Comment:
As discussed with Frank, this is not a good plan, and we should figure out some way of remedying it.

This was how the plan was produced:
* First, the filter to join step noticed that there is a filter on equality of `b` to `'this'`, so it changed the filter to a join on the foo(b) index (a.k.a `ArrangeBy{keys: [[#1]], Get {...(u1)}`)  + `this`. 
* At some point the two joins got fused.
* Then the join implementation determination step replaced the `ArrangeBy{keys: [[#1]], Get {...(u1)}` with an `ArrangeBy` on something that is not already an index.